File: api/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/api/paging/Page.scala
/*
 * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
 */
package com.lightbend.lagom.scaladsl.api.paging

import com.lightbend.lagom.internal.api.paging.CorePage

/**
 * Captures paging information.
 */
final class Page private (private val delegate: CorePage) {
  def pageNo: Option[Int] = delegate.pageNo
  def pageSize: Option[Int] = delegate.pageSize

  override def equals(that: Any): Boolean = {
    if (this == that) true
    else that match {
      case that: Page => delegate == that.delegate
      case _          => false
    }
  }

  override def hashCode: Int = delegate.hashCode
  override def toString: String = delegate.toString
}

object Page {
  /**
   * Factory method for creating `Page` instances.
   */
  def apply(pageNo: Option[Int], pageSize: Option[Int]): Page = new Page(CorePage(pageNo, pageSize))
}

File: api/javadsl/src/main/java/com/lightbend/lagom/javadsl/api/paging/Page.java
/*
 * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
 */
package com.lightbend.lagom.javadsl.api.paging;

import java.util.OptionalInt;

import scala.compat.java8.OptionConverters;

import com.lightbend.lagom.internal.api.paging.CorePage;

/**
 * A page object, use to capture paging information.
 */
public final class Page {
  private final CorePage delegate;

  private Page(CorePage delegate) {
    this.delegate = delegate;
  }

  public Page(OptionalInt pageNo, OptionalInt pageSize) {
    this(CorePage.apply(OptionConverters.toScala(pageNo), OptionConverters.toScala(pageSize)));
  }

  public OptionalInt pageNo() {
    return this.delegate.pageNo().asPrimitive();
  }

  public OptionalInt pageSize() {
    return this.delegate.pageSize().asPrimitive();
  }

  @Override
  public boolean equals(Object o) {
    if (this == o)
      return true;
    if (o == null || getClass() != o.getClass())
      return false;

    Page page = (Page) that;

    return delegate.equals(that.delegate);

  }

  @Override
  public int hashCode() {
    return delegate.hashCode();
  }

  @Override
  public String toString() {
    return delegate.toString();
  }
}
